connector_helpscout/models/helpscout_thread/common.py

```diff
@@ -0,0 +1,119 @@
+            ('movedFromMailbox', 'Moved From Mailbox'),
+            ('merged', 'Merged'),
+            ('imported', 'Imported'),
+            ('workflow', 'Workflow'),
+            ('importedExternal', 'Imported External'),
+            ('changedTicketCustomer', 'Changed Ticket Customer'),
+            ('deletedTicket', 'Deleted Ticket'),
+            ('restoredTicket', 'Restored Ticket'),
+            ('originalCreator', 'Original Creator'),
+        ],
+    )
+    helpscout_action_source = fields.Integer(
+        help='ID associated with HelpScout Action Type.',
+    )
+    helpscout_from_mailbox = fields.Integer(
+        help='If the conversation was moved, this represents the Mailbox from '
+             'which it was moved.',
+    )
+    helpscout_date_opened = fields.Datetime(
+        help='Date that thread was viewed by the customer.',
+    )
+    helpscout_type = fields.Selection(
+        selection=[
+            ('lineitem', 'Line Item'),
+            ('note', 'Note'),
+            ('message', 'Message'),
+            ('chat', 'Chat'),
+            ('customer', 'Customer'),
+            ('forwardparent', 'Forward Parent'),
+            ('forwardchild', 'Forward Child'),
+            ('phone', 'Phone'),
+        ],
+    )
+    helpscout_saved_reply_id = fields.Integer(
+        help='ID of Saved reply that was used to create this Thread.',
+    )
+    helpscout_source_person_type = fields.Selection(
+        selection=[
+            ('customer', 'Customer'),
+            ('user', 'User'),
+        ],
+        help='The person type by whom this thread was created.',
+    )
+    helpscout_source_type = fields.Selection(
+        selection=[
+            ('email', 'Email'),
+            ('web', 'Web'),
+            ('notification', 'Notification'),
+            ('emailfwd', 'Email Forward'),
+            ('api', 'API'),
+            ('chat', 'Chat'),
+        ],
+        help='The method from which this thread was created.',
+    )
+    helpscout_state = fields.Selection(
+        selection=[
+            ('published', 'Published'),
+            ('draft', 'Draft'),
+            ('underreview', 'Under Review'),
+            ('hidden', 'Hidden'),
+        ],
+    )
+    helpscout_to = fields.Char(
+        string='HelpScout Email To',
+        help='Email addresses in the "to" field of the thread email.',
+    )
+    odoo_id = fields.Many2one(
+        string='Message',
+        comodel_name='mail.message',
+        required=True,
+        ondelete='cascade',
+    )
+
+
+class HelpScoutThreadAdapter(Component):
+    """Utilize the API in context."""
+    _name = 'helpscout.thread.adapter'
+    _inherit = 'helpscout.adapter'
+    _apply_on = 'helpscout.thread'
+    _helpscout_endpoint = 'Conversations'
+
+    def create(self, data, conversation_id):
+        return self.endpoint.create_thread(conversation_id,
+                                           self.new_record(data))
+
+    def write(self, _id, data, conversation_id):
```

Comment:
Hmmm you're going to have a hard time injecting the extra arg here I think. Is the conversation ID not already encapsulated within the data? I feel like it should be.

Reply:
It's not in the thread object, unfortunately (https://developer.helpscout.com/help-desk-api/objects/thread/).

These methods are also out of scope since this PR doesn't include exporting, so I'll remove them for now.